### PR TITLE
feat: new wmill CLI

### PR DESCRIPTION
**Pull request by HurricanKai**
Basic CLI tool. Main features for now are syncing local files to a remote. Can handle multiple different remotes with different logins. No OAuth support.
Currently made to be published to deno.land as `windmill`, allowing automatic updates via `windmill upgrade` & installation via `deno install --allow-net --allow-read --allow-write --allow-env --unstable https://deno.land/x/windmill/main.ts`.
Includes command completions via cliffy.

TODO:
- [x] change to only accept *.\<type\>.json files, (ie test.flow.json)
- [x] add more context global options (--token, --username + --password)
  - [x] Rework all commands to use a function to aquire execution context
  - [x] Add --token
  - [x] Add --username --password
- [x] add "remote" subcommand to manage different remotes
  - [x] `remote add`
  - [x] `remote remove`
  - [x] `remote` (implicit list)
  - [x] use this when determining execution context
  - [x] save default remote
- [x] add "user" subcommand to manage users (add/remove)
  - [x] `user add`
  - [ ] `user remove`
  - [x] `user` (implicit list)  
- [x] add setup command to setup default url/email/password/workspace
- [x] variable subcommand
  - [x] `variable push`
  - [x] `variable` (implicit list)
- [x] workspace subcommand
  - [x] `workspace get-default`
  - [x] `workspace set-default`
  - [x] `workspace` (implicit list)
- [x] script subcommand
  - [x] `script push`
  - [x] `script show`
  - [x] `script` (implicit list)
- [x] resource subcommand
  - [x] `resource push`
  - [x] `resource` (implicit list)
- [x] flow subcommand
  - [x] `flow push`
  - [x] `flow` (implicit list)
- [x]  `login`
- [x] root push command

### CODE REVIEW

File: cli/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+A simple CLI allowing interactions with windmill from the command line.
+
+## Login
+
+Logging in will save a token to your local computer, into `~/.config/windmill/<hash>/token` (or `C:\Users\<username>\AppData\Roaming\windmill\<hash>\token` on windows).
+This is inherently unsafe, so do not log into the CLI on untrusted devices.
+
+## Completion
+
+The CLI comes with completions out of the box via `windmill completions <shell>`. (Via [cliffy](https://cliffy.io/))
```
Comment (rubenfiszel) [inline cli/README.md]: :heart_eyes:

File: cli/main.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import workspace from "./workspace.ts";
+import resource from "./resource.ts";
+import remote from "./remote.ts";
+import user from "./user.ts";
+import setup from "./setup.ts";
+import variable from "./variable.ts";
+import push from "./push.ts";
+
+await new Command()
+  .name("windmill")
```
Comment (rubenfiszel) [inline cli/main.ts]: wmill

File: cli/main.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  )
+  .globalOption(
+    "--password <password:string>",
+    "Specify credentials to use for authentication. This will not be stored. It will only be used to exchange for a token with the API server, which will not be stored either.",
+    {
+      depends: ["email"],
+      conflicts: ["token"],
+    }
+  )
+  .version("v0.0.0")
```
Comment (rubenfiszel) [inline cli/main.ts]: we might want to not include version or if possible use the one from version.txt at the root

Comment (HurricanKai) [reply cli/main.ts]: I don't think deno has a way to read the version "compile" time.
We'd have to set it manually.
~~Could also just put 1.0.0 and then version the CLI separately.~~ (nevermind, the deno version aligns with the git tag...)

**Pull request merged**